src/loops/programs_with_repetition/hard/q4/ArtificialSelection.java

```diff
@@ -0,0 +1,39 @@
+package loops.programs_with_repetition.hard.q4;
+
+/*
+Write a program that accepts a sentence from the user and prints the sentence after reversing every other word,
+starting with the first word, using nested loops. Assume that every word is separated by a single space " ",
+and there are no consecutive spaces.
+
```

Comment:
What would you think about adding a hint that there should be a space before the first word? I think it is not necessarily obvious here.

Reply:
I just worry that people would get quite stuck without it, and they might assume our test contains a space at the start.

Reply:
Or go about it logically in another way which would make it difficult to pass our test, even if they went about it correctly.

Reply:
I don't know, the way I see it, even if the students go about it a different way the test should still work, that's why we treat java questions like a black box we can't see into: put something in and see what comes out without seeing the process. An easy way to get around the first space is for them to just flip the first word outside the loops and then start the loops after in my opinion, but admittedly I haven't tried to solve the question myself. I don't see why they would assume our test contains a space at the start.

Reply:
I was just worried that someone could assume that if every word is separated by space, they might think that includes having one in front of the first word. However, that is probably an edge case, so